### PR TITLE
fix output transforms

### PR DESCRIPTION
**Pull request by kashif**
Output transforms should now work for pytorch and mxnet. Also fixed the output type of `CDFtoGaussianTransform` from `float64` to `float32` by default.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by lostella — CHANGES_REQUESTED** [review]
I only have a minor comment, otherwise this looks good!

File: src/gluonts/mx/component.py [hunk 1]
```diff
@@ -131,3 +138,8 @@ def equals_representable_block(
 @skip_encoding.register(mx.gluon.ParameterDict)
 def skip_encoding_mx_gluon_parameterdict(v: mx.gluon.ParameterDict) -> bool:
     return True
+
+
+@tensor_to_ndarray.register(mx.ndarray.NDArray)
+def _(tensor: mx.ndarray.NDArray) -> np.ndarray:
+    return tensor.asnumpy()
```
[the remark this answers is not in the document]
Comment (kashif) [reply src/gluonts/mx/component.py:145]: Yes you are right... I’ll change it in the morning, thanks!

**Comment by kashif** [comment]
@lostella all done on my side. With this PR I would have everything I need to use PyTorch with glounts as a dependency for my work. The last remaining issue is that on PyTorch the `DataLoaders` do not work when `num_workers` is more than zero on the GPU since pytorch's cuda context doesn't work with `fork`ing. My solution has been to just define an iterative dataset and then use pytorch's dataloader instead and that seems to work. 

Do you think it would be possible to make a release a new point version of gluonts with the changes up till now? Or should o make a PR for the dataloader proposal? Thanks!

**Comment by lostella** [comment]
@kashif thank you! A new release should come pretty soon, I think #1196 is the only really blocking story. The other stories in the [milestone](https://github.com/awslabs/gluon-ts/milestones/v0.7) we may punt to a next release in case they cannot be addressed shortly.

Regarding the data loader problem, I'm not sure I understand the issue. One thing I know, is that on some platforms one needs to enforce the *spawn* method in order to use multi-processing with GluonTS' data loader.

Maybe we can move these questions to a dedicated [discussion topic](https://github.com/awslabs/gluon-ts/discussions)? Any improvement to the data loader is welcome -- this said, it should be possible to the trainer with whatever source of batches one is comfortable with, including PyTorch's or fully custom iterators, as well as the other way around (GluonTS data transformation & loading utils with a custom training procedure).

**Review by lostella — APPROVED** [review]

**Pull request merged**